Lib/pydoc.py

```diff
@@ -1910,7 +1911,11 @@ def interact(self):
                 if not request: break
             except (KeyboardInterrupt, EOFError):
                 break
-            request = replace(request, '"', '', "'", '').strip()
+            request = request.strip()
+            if not (len(request) == 2 and
```

Comment:
I suggest to replace the code with:
```python
            if (len(request) > 2 and request[0] == request[-1] in ("'", '"')
                    and request[0] not in request[1:-1]):
                request = request[1:-1]
```
(and add a comment).

Reply:
Do you think it's okay to keep a quotation mark in cases like:
"spam
eggs"
?

We are catching these "typos" now, and I believe this can be useful to prevent errors produced by copy and pasting, when quotation marks might be added accidentally.

Reply:
If quotation marks are added accidentally, they should added at both sides and be different from quotations inside. I don't think that we should accept arbitrary garbage.

The current code misbehave not only with `r'`, but with `'` and `"""`. And your current patch doesn't fixes the latter two cases.

Reply:
Yes, I believe you are right. I'll update the PR to address this :)